activesupport/lib/active_support/core_ext/string/filters.rb

```diff
@@ -64,7 +64,7 @@ def remove!(*patterns)
   #   'And they found that many people were sleeping better.'.truncate(25, omission: '... (continued)')
   #   # => "And they f... (continued)"
   def truncate(truncate_at, options = {})
-    return dup unless length > truncate_at
+    return dup.freeze unless length > truncate_at
```

Comment:
I think it should be the opposite, we should always return a non-frozen string. In the same way as `gsub` and `strip` don't. I believe it only return frozen strings because of the `frozen_string_literal` pragm in this file and because we are using interpolation before returning.

Reply:
```
>> 'foo'.freeze.gsub('a', '').frozen?
=> false
>> 'foo'.freeze.gsub('o', '').frozen?
=> false
```

Reply:
Ah right on, I think that's even more predictable. I wrongly interpreted that the `frozen_string_literal` thing was trying to ensure returns were frozen.

I'll swap it.